tests/classifiers/test_keras.py

```diff
@@ -397,12 +397,171 @@ def test_repr(self):
+                       loss_gradient_expected, _from_logits=True)
+
+        # =============================== #
+        # sparse_categorical_crossentropy #
+        # =============================== #
+
+        loss_name = 'sparse_categorical_crossentropy'
+
+        # loss_gradient should be the same for probabilities and logits but dependent on loss function
+
+        loss_gradient_expected = np.asarray([0.0, 0.0, 0.0, -0.09573442, -0.0089094, 0.01402334, 0.0258659,
+                                             0.08960329, 0.10324767, 0.10624839, 0.06578761, -0.00018638,
+                                             -0.01345262, -0.08770822, -0.04990875, 0.04288402, -0.06845165,
+                                             -0.08588978, -0.08277036, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+
+        # testing with probabilities
+
+        for loss_type in ['label', 'function_losses', 'function_backend']:
+            logger.info('loss_name: {}, loss_type: {}, output: probabilities', loss_name,
+                        loss_type.format(loss_name, loss_type))
+
+            _run_tests(loss_name, loss_type, y_test_pred_expected, class_gradient_probabilities_expected,
+                       loss_gradient_expected, _from_logits=False)
+
+        # testing with logits
+
+        for loss_type in ['function_backend']:
+            logger.info('loss_name: {}, loss_type: {}, output: logits', loss_name,
+                        loss_type.format(loss_name, loss_type))
+
+            _run_tests(loss_name, loss_type, y_test_pred_expected, class_gradient_logits_expected,
+                       loss_gradient_expected, _from_logits=True)
+
+        # =========================== #
+        # kullback_leibler_divergence #
+        # =========================== #
+
+        loss_name = 'kullback_leibler_divergence'
+
+        # loss_gradient should be the same for probabilities and logits but dependent on loss function
+
+        loss_gradient_expected = np.asarray([0.0, 0.0, 0.0, -0.09573442, -0.0089094, 0.01402334, 0.0258659, 0.08960329,
+                                             0.10324767, 0.10624839, 0.06578761, -0.00018638, -0.01345262, -0.08770822,
+                                             -0.04990875, 0.04288402, -0.06845165, -0.08588978, -0.08277036, 0.0, 0.0,
+                                             0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+
+        # testing with probabilities
+
+        for loss_type in ['function_losses']:
+            logger.info('loss_name: {}, loss_type: {}, output: logits', loss_name,
+                        loss_type.format(loss_name, loss_type))
+
+            _run_tests(loss_name, loss_type, y_test_pred_expected, class_gradient_probabilities_expected,
+                       loss_gradient_expected, _from_logits=False)
+
+        # # ================= #
+        # # cosine_similarity #
+        # # ================= #
+        #
+        # loss_name = 'cosine_similarity'
```

Comment:
Should this test stay commented?

Reply:
I will remove this loss function for now because its name changes between Keras/TF versions.